File: interval-server/src/main/java/com/interval/auth/AuthService.java
package com.interval.auth;

/**
 * 认证服务接口
 * 
 * 定义用户认证相关的业务逻辑
 */
public interface AuthService {
    
    /**
     * 用户登录
     * 
     * @param username 用户名
     * @param password 密码
     * @return 登录响应（包含 JWT token）
     * @throws IllegalArgumentException 如果用户名或密码错误
     */
    LoginResponseDto login(String username, String password);
    
    /**
     * 用户注册
     * 
     * @param username 用户名
     * @param password 密码
     * @return 注册响应（包含用户信息）
     * @throws IllegalArgumentException 如果用户名已存在或密码不符合要求
     */
    RegisterResponseDto register(String username, String password);
}

File: interval-client/src/services/authService.ts
import { http } from '@/services/http';
import type {
  ChangePasswordRequest,
  LoginRequest,
  LoginResponseDto,
  RegisterRequest,
  RegisterResponseDto,
} from '@/types/auth';

export async function login(username: string, password: string): Promise<LoginResponseDto> {
  return http.post<never, LoginResponseDto>('/api/auth/login', { username, password } satisfies LoginRequest);
}

export async function register(username: string, password: string): Promise<RegisterResponseDto> {
  return http.post<never, RegisterResponseDto>('/api/auth/register', { username, password } satisfies RegisterRequest);
}

export async function changePassword(request: ChangePasswordRequest): Promise<void> {
  return http.post<never, void>('/api/auth/change-password', request);
}
